apps/explorer/src/hooks/useSearch.ts

```diff
@@ -120,6 +122,35 @@ const getResultsForAddress = async (
     ];
 };
 
+async function addressExists(client: IotaClient, address: string): Promise<boolean> {
+    const normalized = normalizeIotaObjectId(address);
+    if (!isValidIotaAddress(normalized)) return false;
+
+    try {
+        const fromTransactions = await client.queryTransactionBlocks({
+            filter: { FromAddress: normalized },
+            limit: 1,
+        });
+        if (fromTransactions.data.length > 0) return true;
+
+        const toTransactions = await client.queryTransactionBlocks({
+            filter: { ToAddress: normalized },
+            limit: 1,
+        });
+        if (toTransactions.data.length > 0) return true;
```

Comment:
Wouldn't this also work by doing:

```suggestion
const transactions = await client.queryTransactionBlocks({
    filter: { FromOrToAddress: { addr: normalized } },
    limit: 1,
});
if (toTransactions.data.length > 0) return true;
```

?